### PR TITLE
feat: add support for specifying wallet derivation account and index values

### PR DESCRIPTION
**Pull request by antbern**
## Motivation and Context

We need to support having multiple accounts and addresses with the same mnemonic :rocket:

## Summary

Provide a short summary of changes in this pull request.

- Introduce `MnemonicDerivationOption` struct with the `account` and `index` numbers.
- Use the `account` and `index` numbers in the key derivation for both EVM and IOTA Rebased.
- Update tests.
- Add binding for WASM (`setWalletAccount(account,index)`) and add example usage in `generate_new_address` 🔥 

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [x] Code follows project style and guidelines.
- [x] No redundant or duplicate code.
- [x] No added new dependencies without clear justification and approval.

### Documentation

- [x] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [x] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [x] Tests pass locally.

### Build & CI/CD

- [x] Confirmed no CI/CD pipeline issues.
- [x] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

**Review by kristifidani — APPROVED** [review]
Great! LGTM 💯 🚀

**Review by hauju — APPROVED** [review]
Just a comment that not fits

File: bindings/wasm/src/lib.rs [hunk 1]
```diff
@@ -549,6 +549,21 @@ impl ETOPaySdk {
             .map_err(|e| format!("{e:#?}"))
     }
 
+    /// Set the account and index used for deriving the private key from the mnemonic. This is used
+    /// to have multiple accounts and addresses with a single mnemonic.
+    ///
+    /// @param {number} start - The start page
+    /// @param {number} limit - The limit per page
```
[the remark this answers is not in the document]
Comment (antbern) [reply bindings/wasm/src/lib.rs:556]:
```suggestion
    /// @param {number} account - The account number
    /// @param {number} index - The index number
```
